src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,538 @@
+      await expect(iconLocator).toHaveCSS("color", color);
+    });
+  });
+
+  colorData.forEach((backgroundColor) => {
+    test(`should check background color as ${backgroundColor}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent bg={backgroundColor} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("background-color", backgroundColor);
+    });
+  });
+
+  colorData.forEach((iconColor) => {
+    test(`should check icon color as ${iconColor}`, async ({ mount, page }) => {
+      await mount(<IconComponent color={iconColor} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("color", iconColor);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 24],
+    [SIZE.MEDIUM, 32],
+    [SIZE.LARGE, 40],
+    [SIZE.EXTRALARGE, 56],
+  ] as [IconProps["fontSize"], number][]).forEach(
+    ([fontSize, heightAndWidth]) => {
+      test(`should check height and width as ${heightAndWidth} when fontSize is set as ${fontSize}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<IconComponent fontSize={fontSize} />);
+
+        const iconLocator = await page.getByTestId("icon");
+        await expect(iconLocator).toHaveCSS("height", `${heightAndWidth}px`);
+        await expect(iconLocator).toHaveCSS("width", `${heightAndWidth}px`);
+      });
+    }
+  );
+
+  ([
+    ["circle", 50],
+    ["rounded-rect", 20],
+    ["square", 0],
+  ] as [IconProps["bgShape"], number][]).forEach(([bgShape, radius]) => {
+    test(`should check bgShape as ${bgShape} when radius is set as ${radius}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent bgShape={bgShape} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS(
+        "border-bottom-left-radius",
+        `${radius}%`
+      );
+      await expect(iconLocator).toHaveCSS(
+        "border-bottom-right-radius",
+        `${radius}%`
+      );
+      await expect(iconLocator).toHaveCSS(
+        "border-top-left-radius",
+        `${radius}%`
+      );
+      await expect(iconLocator).toHaveCSS(
+        "border-top-right-radius",
+        `${radius}%`
+      );
+    });
+  });
+
+  ([
+    [SIZE.EXTRASMALL, 16],
+    [SIZE.SMALL, 24],
+    [SIZE.MEDIUM, 32],
+    [SIZE.LARGE, 40],
+    [SIZE.EXTRALARGE, 56],
+  ] as [IconProps["bgSize"], number][]).forEach(([size, heightAndWidth]) => {
+    test(`should check bgSize as ${size} when height and width is set as ${heightAndWidth}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent bgSize={size} />);
+
+      const iconLocator = await page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("height", `${heightAndWidth}px`);
+      await expect(iconLocator).toHaveCSS("width", `${heightAndWidth}px`);
+    });
+  });
+
+  ([
+    [["left"], "left", "bottom", 0, 0],
```

Comment:
**suggestion:** I _think_ this syntax should work. 
```suggestion
    ["left", "left", "bottom", 0, 0],
    ["top", "top", "bottom", 0, 0],
    ["left", "left", "top", 0, 120],
    ["bottom", "bottom", "top", 0, 120],
    ["bottom", "bottom", "left", 0, 120],
    ["bottom", "bottom", "right", 0, 120],
    ["top", "top", "left", 0, 0],
    ["top", "top", "right", 0, 0],
    ["right", "right", "bottom", 700, 0],
    ["right", "right", "top", 700, 120],
```

Reply:
tooltipFlipOverrides test won't work if I remove [] as this prop needs to pass with an array here
<img width="728" alt="Screenshot 2023-10-03 at 09 50 56" src="https://github.com/Sage/carbon/assets/88449494/bf200ff3-8f41-439d-a9e3-e6cad9377189">

Reply:
Ignore me 😆 I thought it used the same format as the `tooltipPosition` which accepts just a string.